ChopChop/ViewModels/Recipes/RecipeViewModel.swift

```diff
@@ -41,32 +41,30 @@ final class RecipeViewModel: ObservableObject {
                 }
             }
 
-        if let id = recipe?.parentOnlineRecipeId {
-            parentRecipeCancellable = storageManager.onlineRecipePublisher(id: id)
-                .sink(receiveCompletion: { completion in
-                    switch completion {
-                    case .finished:
-                        break
-                    case .failure:
-                        self.parentRecipe = nil
-                    }
-                }, receiveValue: { value in
-                    self.parentRecipe = value
-                })
+        guard let parentId = recipe?.parentOnlineRecipeId else {
+            return
+        }
+        storageManager.fetchOnlineRecipe(id: parentId) { onlineRecipe, _ in
+            self.parentRecipe = onlineRecipe
         }
```

Comment:
Does this mean that if the network is slow, the recipe view will take a long time to load? It seems like this method isn't asynchronous, unlike the previous publisher.

Reply:
the firebase operations are all asynchronous, so that closure part is like a function that is passed as a param, and only called when firebase finishes the api call